src/main/java/com/hubspot/jinjava/lib/tag/eager/EagerSetTag.java

```diff
@@ -133,6 +133,21 @@ public String getEagerTagImage(TagToken tagToken, JinjavaInterpreter interpreter
         )
       );
     }
+    if (
+      eagerExecutionResult.getResult().isFullyResolved() &&
+      interpreter.getContext().isDeferredExecutionMode()
+    ) {
+      try {
+        getTag()
+          .executeSet(
```

Comment:
Try to execute the set after waterfall-deferring if it is fully resolvable so that we can use the value during this context